framework/include/materials/MaterialPropertyInterface.h

```diff
@@ -27,6 +27,12 @@ InputParameters validParams();
 template <>
 InputParameters validParams<MaterialPropertyInterface>();
 
+#define adDeclareADProperty(Type) this->template declareADProperty<Type>
```

Comment:
If you just leave the template pieces out of these declarations - then users don't have to use the weird `()` stuff.

Also: I would rather name _these_ the normal name (i.e. `declareADProperty` and name the actual "real" function something else.  These are going to be the things that users actually interact with.

Reply:
That goes for all of these kinds of interfaces.  What we want to get to is something that looks nearly identical to the non-AD code.